autoconfig: prevent a data race in TestAutoConfig

knz opened the pull request:
Needed for #101069.

The calls to Peek and Pop can run concurrently.

Release note: None
Epic: CRDB-23559

knz committed:
autoconfig: prevent a data race in TestAutoConfig

The calls to Peek and Pop can run concurrently.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/101076)
<!-- Reviewable:end -->

knz commented:
bors r-

knz committed:
autoconfig: properly order versions in TestAutoConfig

Before this change, the test was kicking off tasks concurrently with
version upgrades. To reduce noise in logs, this patch fixes it to make
it wait until upgrades have completed.

Release note: None

knz commented:
bors r=adityamaru

knz committed:
autoconfig: transactional SQL should wait on leases

Release note: None

knz commented:
bors r=adityamaru
